lib/apis/auth_api.dart

```diff
@@ -0,0 +1,89 @@
+// supabase database password : jeksov-Sifnag-hyjzy9
+
+import 'package:fitness_tracker_app/core/core.dart';
+import 'package:fitness_tracker_app/core/failure.dart';
+import 'package:fitness_tracker_app/core/providers.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:fpdart/fpdart.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+final authProvider = Provider((ref) {
+  final authApi = ref.watch(supabaseAccountProvider);
+  return AuthApi(account: authApi);
+});
+
+abstract class IAuthApi {
+  FutureEither<User> signup(
+      {required String email, required String password, required String name});
+
+  FutureEither<Session> login(
+      {required String email, required String password});
```

Comment:
Is your linter broken? I'd assume it would complain about missing commas here.
```suggestion
  FutureEither<User> signup({
    required String email,
    required String password,
    required String name,
  });

  FutureEither<Session> login({
    required String email,
    required String password,
  });
```